apps/cowswap-frontend/src/common/pure/InlineBanner/index.tsx

```diff
@@ -1,55 +1,78 @@
 import { ReactNode } from 'react'
 
-import { lighten, darken, transparentize } from 'polished'
-import SVG from 'react-inlinesvg'
-import styled, { useTheme } from 'styled-components/macro' // import useTheme
+import styled from 'styled-components/macro'
 
-import iconInformation from 'legacy/assets/cow-swap/alert-circle.svg'
-import iconAlert from 'legacy/assets/cow-swap/alert.svg'
-import iconDanger from 'legacy/assets/cow-swap/alert.svg'
-import iconSuccess from 'legacy/assets/cow-swap/check.svg'
+import { UI } from 'common/constants/theme'
+import { Icon, IconType } from 'common/pure/Icon'
+import { BannerOrientation } from 'common/pure/InlineBanner/banners'
 
-type BannerType = 'alert' | 'information' | 'success' | 'danger' | 'savings'
+export type BannerType = 'alert' | 'information' | 'success' | 'danger' | 'savings'
 
-interface BannerConfig {
-  colorKey: Exclude<BannerType, 'savings'>
-  icon?: string
+interface ColorEnums {
+  icon?: IconType
   iconText?: string
+  color: UI
+  bg: string
+  text: string
 }
 
-const BANNER_CONFIG: Record<BannerType, BannerConfig> = {
-  alert: {
-    icon: iconAlert,
-    colorKey: 'alert',
-  },
-  information: {
-    icon: iconInformation,
-    colorKey: 'information',
-  },
-  success: {
-    icon: iconSuccess,
-    colorKey: 'success',
-  },
-  savings: {
-    iconText: '💸',
-    colorKey: 'success',
-  },
-  danger: {
-    icon: iconDanger,
-    colorKey: 'danger',
-  },
+function getColorEnums(bannerType: BannerType): ColorEnums {
```

Comment:
Now the code is unsafe.
I can delete of the cases and there is no typescript error.
`Record<>` gives us protection.

<img width="675" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/4bd9bf01-1ca0-4087-8cdf-581c30182a2a">

Reply:
Fixed